geomstats/geometry/product_hpd_matrices_and_siegel_disks.py

```diff
@@ -0,0 +1,111 @@
+"""The ProductHPDMatricesAndSiegelDisks manifold.
```

Comment:
the filename is too long: `product_hpd_and_siegel` instead? the class can then keep its name.

Reply:
Hello @lpereira95,
Is `product_hpd_and_siegel_disks` fine?
Or is it still too long?
If possible, I would like to show that there are several Siegel spaces.

Reply:
Hello @lpereira95,
I have changed `product_hpd_matrices_and_siegel_disks` into `product_hpd_and_siegel_disks` in the file names.
Tell me if it is still too long.

Reply:
Done.